TLM/TLM/Manager/Impl/VehicleBehaviorManager.cs

```diff
@@ -335,53 +350,6 @@ public class VehicleBehaviorManager : AbstractCustomManager, IVehicleBehaviorMan
 							vehicleData.m_waitCounter = 0;
 						}
 
-                        // Check if turning in the preferred direction, and if turning while it's red is allowed
-                        if (Options.turnOnRed) {
-                            if (JunctionRestrictionsManager.Instance.IsTurnOnRedAllowed(prevPos.m_segment, isTargetStartNode)) {
-                                // Check if vehicle has stopped
-                                if ((vehicleState.JunctionTransitState == VehicleJunctionTransitState.Stop && sqrVelocity <= TrafficPriorityManager.MAX_SQR_STOP_VELOCITY)
-                                    || isRecklessDriver && sqrVelocity <= TrafficPriorityManager.MAX_YIELD_VELOCITY) {
-
-                                    ushort uCurrentSegment = prevPos.m_segment;
-                                    ushort uTargetSegment = position.m_segment;
-
-                                    // If you can turn preferred and you're not going straight, continue
-                                    SegmentGeometry currentSegGeo = SegmentGeometry.Get(uCurrentSegment);
-                                    SegmentEndGeometry currentSegEndGeo = currentSegGeo.GetEnd(targetNodeId);
-#if DEBUG
-                                    if (debug)
-                                        Log._Debug($"VehicleBehaviorManager.MayChangeSegment({frontVehicleId}): uCurrentSegment={uCurrentSegment}, hasValidTurnOnRedOutgoingSegment={currentSegGeo.HasValidTurnOnRedOutgoingSegment(currentSegEndGeo)}, isStraightSegment1={currentSegGeo.IsStraightSegment(uTargetSegment, false)}, isStraightSegment2={currentSegGeo.IsStraightSegment(uTargetSegment, true)}");
-#endif
-                                    if (currentSegGeo.HasValidTurnOnRedOutgoingSegment(currentSegEndGeo) && !currentSegGeo.IsStraightSegment(uTargetSegment, false) && !currentSegGeo.IsStraightSegment(uTargetSegment, true)) {
```

Comment:
I removed the expensive `HasValidTurnOnRedOutgoingSegment` check because JunctionRestrictionManager should guarantee that `IsTurnOnRedAllowed` is only true if  `HasValidTurnOnRedOutgoingSegment` is true.

Reply:
I also removed `!currentSegGeo.IsStraightSegment(uTargetSegment, false) && !currentSegGeo.IsStraightSegment(uTargetSegment, true)` because we don't need to guess whether the node is the start or end node.